### DISCUSSION TITLE
Running MT3DMS without re-running MODFLOW

### DISCUSSION THREAD

**Discussion by mahaderony** [modflowpy/flopy#1576] [Q&A] [answered]
Is there any way to run MT3DMS by just linking a MODFLOW simulation? I am running a loop with MT3DMS and this would save me time.

**Comment by emorway-usgs** [comment, marked as answer]
@mahademrony Could you be a little more specific?  The FMI package in MT3DMS enables reading of a MODFLOW "linker" file.  You need to activate the LMT package within MODFLOW-2005 in order to generate the linker file for MT3DMS to read.

**Reply by mahaderony** [reply]
I want to run MT3DMS in a loop for different input parameter. The MODFLOW model that I am running is in NWT and I do have the "link' file. I want to know how I can initiate the MT3DMS simulation from the link file. What would be the syntax.

**Reply by emorway-usgs** [reply]
It's still not very clear what you are seeking to do.  If you want to evaluate the effect of different input parameter values on the output, why not use [PEST](https://github.com/usgs/pestpp)?  If you are asking "how do I change values written to the linker file and rerun MT3DMS?", safest to rerun NWT with your updated flow model parameters and then rerun MT3DMS.

**Reply by mahaderony** [reply]
I'll try to be more clear, 
I have a local MODFLOW model that runs on modflow-nwt to solve the flow equations. 
I am running MT3DMS on this model
now, I want to run 5 different MT3DMS model for 5 different dispersivity values.
I am using flopy to load my existing MODFLOW_NWT model and write all necessary packages. 
Now my question is, can I run MT3DMS by only importing the "link" file from my modflow simulation?

**Reply by emorway-usgs** [reply]
Once the linker file is generated, that's all you need from the flow model to rerun MT3DMS as many times as you like.  If you want to explore 5 alternative sets of dispersivity, all using the same flow model set up and do it with flopy, then you shouldn't need to reload the flow model into flopy.  Instead, you will want to load the _MT3DMS_ model into flopy.  Once MT3DMS has been loaded into flopy, you can alter the dispersivities as you like and rerun MT3DMS (using flopy) and retrieve whichever output you like.  With this approach, you should only need to load MT3DMS into memory once, though you'll end up writing it 5 times - each time will write the model to disk with one of your 5 pre-specified dispersivity arrays.

**Reply by mahaderony** [reply]
Thank you. but I am getting the following error

"Exception: BTN error. Required input is None, but no modflow model. If no modflow model is passed to Mt3dms, then values must be specified in the BTN constructor for: nlay, nrow, ncol, nper, laycon, delr, delc, htop, dz, perlen, nstp, and tsmult."

Here is my code for loading the MT3DMS

```
modelname = 'Carrier_MT3DMS'
mt = flopy.mt3d.Mt3dms(modelname, 
                       exe_name = 'mt3dms5b.exe', 
                       ftlfilename='mt3d_link.ftl', 
                      )
```
so I have to re-define the BTN package for the entire model?

**Reply by emorway-usgs** [reply]
Oh right, I forgot about passing the flow model to the MT3D model instantiation so that the discretization parameters can be auto-filled using the information in the flow model object.  If you load your flow model into `gwf`, for example, then you should be able to pass the flow model's discretization information to the MT3DMS instantiation (inside flopy) by passing the flow model object with (note the last argument passed):

```
modelname = 'Carrier_MT3DMS'
mt = flopy.mt3d.Mt3dms(
    modelname=modelname,
    exe_name='mt3dms5b.exe',
    ftlfilename='mt3d_link.ftl', 
    modflowmodel=gwf  
)
```

**Reply by mahaderony** [reply]
It works! Thank you so much!!
